### ISSUE TITLE
4gpus with total batchsize=8 and 1gpu with batchsize=8, same lr, different results

### ISSUE THREAD

**Issue by Fangyh09** [horovod/horovod#1619]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.0.0
3. Horovod version: 0.15.2

I think the two experiments should give same results, but the results are quite different. I set lr = 1e-4 to both of the two experiments. What are the possible reasons?  Currently, I think it may be related to `torch.optim.Adam(model.parameters(), float(args.lr * hvd.size())...`.  Thanks !

1. 4gpus with total batchsize=8
2. 1gpu with batchsize=8

Related codes:
```python
optimizer = torch.optim.Adam(model.parameters(), float(args.lr * hvd.size()),
                                 weight_decay=args.weight_decay)

for epoch in range(args.start_epoch, args.epochs):
    train_sampler.set_epoch(epoch)
    adjust_learning_rate(args, optimizer, epoch)

def adjust_learning_rate(args, optimizer, epoch):
    lr = args.lr * (0.1 ** (epoch // 5))

    for param_group in optimizer.param_groups:
        param_group['lr'] = lr
```

**Comment by tgaddair** [comment]
Hey @Fangyh09, can you clarify something about your experiment setup: it sounds like you're keeping the effective batch size the same across both experiments, but scaling the learning rate, is that correct?

Specifically:

- 1 GPU: 8 local batch size -> 8 total batch size, 1e-4 learning rate
- 4 GPUs: 2 local batch size -> 8 total batch size, 4e-4 learning rate

If my assumption is correct, then the problem is that you are scaling the learning rate without scaling the total batch size.  You need to either scale up the total batch size (8 local batch size -> 32 total batch size, 4e-4 learning rate), or hold the learning rate constant (8 total batch size, 1e-4 learning rate). 

Keeping the local batch size constant and increasing the total batch size is recommended to get the most performance out of your network.  Decreasing the local batch size and keeping the total batch size constant could be useful for making an apples-to-apples comparison between convergence with 1 GPU vs 4 GPUs.

In general, it's important to note that the learning rate needs to scale with the total batch size.

**Comment by Fangyh09** [comment]
I see, I thought it is
```
1 GPU: 8 local batch size -> 8 total batch size, 1e-4 learning rate
4 GPUs: 2 local batch size -> 8 total batch size, 1e-4 learning rate
```
You are right. I should change `args.lr * hvd.size()` to `args.lr`. Thanks !
